scripts/common.py

```diff
@@ -712,3 +715,99 @@ def __enter__(self):
 
     def __exit__(self, etype, value, traceback):
         os.chdir(self.saved_path)
+
+
+@contextlib.contextmanager
+def managed_process(command_args, shell=False, timeout_secs=60, **kwargs):
+    """Context manager for starting and stopping a process gracefully.
+
+    Args:
+        command_args: list(int | str). A sequence of program arguments, where
+            the program to execute is the first item.
+        shell: bool. Whether the command should be run inside of its own shell.
```

Comment:
There are notes that using shell=True is concerning from a security perspective, e.g. https://docs.python.org/3/library/subprocess.html#subprocess.Popen . Should we replicate that here as a warning?

Reply:
Sounds good, done.